corehq/apps/ota/utils.py

```diff
@@ -99,36 +145,17 @@ def is_permitted_to_restore(domain, couch_user, as_user, has_data_cleanup_privel
     :returns: a tuple - first a boolean if the user is permitted,
         secondly a message explaining why a user was rejected if not permitted
     """
-    message = None
-    if couch_user.is_commcare_user() and domain != couch_user.domain:
-        message = u"{} was not in the domain {}".format(couch_user.username, domain)
-    elif couch_user.is_web_user() and domain not in couch_user.domains and not couch_user.is_superuser:
-        message = u"{} was not in the domain {}".format(couch_user.username, domain)
-    elif (couch_user.is_web_user() and
-            couch_user.is_member_of(domain) and
-            as_user is not None):
-        if not has_data_cleanup_privelege and not couch_user.is_superuser:
-            message = u"{} does not have permissions to restore as {}".format(
-                couch_user.username,
-                as_user,
-            )
-
-        try:
-            username = as_user.split('@')[0]
-            user_domain = as_user.split('@')[1]
-        except IndexError:
-            message = u"Invalid to restore user {}. Format is <user>@<domain>".format(as_user)
-
-        else:
-            if user_domain != domain:
-                # In this case we may be dealing with a WebUser
-                user = WebUser.get_by_username(as_user)
-                if user and user.is_member_of(domain):
-                    message = None
-                else:
-                    message = u"{} was not in the domain {}".format(username, domain)
-
-    return message is None, message
+    try:
```

Comment:
nice cleanup. 
nit: I find it more readable to have helper methods defined below where they are first referenced.